xpk.py

```diff
@@ -1127,6 +1142,120 @@ def zone_to_region(zone) -> str:
   return zone_terms[0] + '-' + zone_terms[1]
 
 
+def get_service_account_name(args) -> str:
+  """Get the name for the service account.
+  Args:
+    args: user provided arguments.
+
+  Returns:
+    the name of the service account.
+  """
+  return f'{args.project}-{_XPK_SERVICE_ACCOUNT}@{args.project}.iam.gserviceaccount.com'
+
+def check_if_service_account_exists(args) -> bool:
+  """Check if a service account with the given name exists in the project.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    True if service account exist, False otherwise.
+  """
+  service_account_name = get_service_account_name(args)
+  command = (
+    f'gcloud iam service-accounts describe {service_account_name}'
+  )
+  return_code = run_command_with_updates(command, 'Service Account Describe', args, verbose=False)
+  if return_code != 0:
+    xpk_print(f'Service Account Describe did not find the service account {service_account_name}.')
+    return False
+  return True
+
+def create_service_account(args) -> int:
+  """Creates a service account in the project.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  command = (
+    f'gcloud iam service-accounts create {args.project}-{_XPK_SERVICE_ACCOUNT} \
+    --description="Service Account for XPK" \
+    --display-name="{args.project}-{_XPK_SERVICE_ACCOUNT}"'
+  )
+  return_code = run_command_with_updates(command, 'Service Account Create', args)
+  if return_code != 0:
+    xpk_print(f'Service Account Create request returned ERROR {return_code}')
+    xpk_print('Make sure you have Service Account Admin Role attached to your user.')
```

Comment:
Can you add a print statement that says: xpk_print('Please look at https://github.com/google/xpk/blob/main/README.md#permission-issues-requires-one-of-permission_name-permissions for how to add permissions to your user.')

Reply:
For now, I haven't added any documentation related to service account in README. So, directing users to that link won't help. I am thinking to update the README once make this functionality default for everyone. wdyt?

Reply:
Sounds good! We have a section in the readme already for adding new permissions to the user. We can extend that / rewrite it for service accounts

Reply:
yes, have an AI on me to update the README once we are ready to release this feature.